File: Java/src/MaximumSubArrayTest.java
import org.junit.Assert;
import org.junit.Test;

public class MaximumSubArrayTest {

    @Test
    public void testMaxSubArray() {
        MaximumSubArray ms = new MaximumSubArray();

        Assert.assertTrue(ms.maxSubArray(new int[]{1}) == 1);

        Assert.assertTrue(ms.maxSubArray(new int[]{-4, -2}) == -2);

        Assert.assertTrue(ms.maxSubArray(new int[]{1, 0, -1}) == 1);

        Assert.assertTrue(ms.maxSubArray(new int[]{-2, 1, -3, 4, -1, 2, 1, -5, 4}) == 6);
    }

}

File: Kotlin/src/MaximumSubArrayTest.kt
import org.junit.Test

class MaximumSubArrayTest {

    @Test
    fun testMaxSubArray() {
        val ms = MaximumSubArray()

        assert(ms.maxSubArray(intArrayOf(1)) == 1)

        assert(ms.maxSubArray(intArrayOf(-4, -2)) == -2)

        assert(ms.maxSubArray(intArrayOf(1, 0, -1)) == 1)

        assert(ms.maxSubArray(intArrayOf(-2, 1, -3, 4, -1, 2, 1, -5, 4)) == 6)
    }

}